android/CouchbaseLite/src/androidTest/java/com/couchbase/lite/LogTest.java

```diff
@@ -111,6 +118,146 @@ public void testEnableAndDisableCustomLogging() {
+            public void run() {
+                String uuidString = UUID.randomUUID().toString();
+                writeAllLogs(uuidString);
+
+                try {
+                    for (File log : path.listFiles()) {
+                        byte[] encoded = Files.readAllBytes(log.toPath());
+                        String contents = new String(
+                                encoded,
+                                StandardCharsets.US_ASCII
+                        );
+                        assertFalse(contents.contains(uuidString));
+                    }
+                } catch(Exception e) {
+                    fail("Exception during test callback " + e.toString());
+                }
+            }
+        });
+        emptyDirectory(path.getAbsolutePath());
+    }
+
+    @Test
+    public void testFileLoggingReEnableLogging() {
+        final File path = new File(
+                context.getCacheDir().getAbsolutePath(),
+                "testFileLoggingReEnableLogging"
+        );
+        final String logDirectory = emptyDirectory(path.getAbsolutePath());
+        LogFileConfiguration config = new LogFileConfiguration(logDirectory)
+                .setUsePlaintext(true);
+        testWithConfiguration(LogLevel.NONE, config, new Runnable() {
+            @Override
+            public void run() {
+                String uuidString = UUID.randomUUID().toString();
+                writeAllLogs(uuidString);
+
+                try {
+                    for (File log : path.listFiles()) {
+                        byte[] encoded = Files.readAllBytes(log.toPath());
+                        String contents = new String(
+                                encoded,
+                                StandardCharsets.US_ASCII
+                        );
+                        assertFalse(contents.contains(uuidString));
+                    }
+
+                    Database.log.getFile().setLevel(LogLevel.VERBOSE);
+                    writeAllLogs(uuidString);
+
+                    File[] filesExceptDebug = path.listFiles(new FilenameFilter() {
+                        @Override
+                        public boolean accept(File dir, String name) {
+                            return !name.toLowerCase().startsWith("cbl_debug_");
+                        }
+                    });
+
+                    for (File log : filesExceptDebug) {
+                        byte[] encoded = Files.readAllBytes(log.toPath());
+                        String contents = new String(
+                                encoded,
+                                StandardCharsets.US_ASCII
+                        );
+                        assertTrue(contents.contains(uuidString));
+                    }
+
+                } catch(Exception e) {
+                    fail("Exception during test callback " + e.toString());
+                }
+            }
+        });
+        emptyDirectory(path.getAbsolutePath());
+    }
+
+    @Test @Ignore
```

Comment:
ignored since the [updateConfig](https://github.com/couchbase/couchbase-lite-android/blob/1b96876a3fb00a3dcbc634c1cd9778a2a5f6dd85/shared/src/main/java/com/couchbase/lite/FileLogger.java#L100) method is not passing in the user agent info. 

[native_c4.cc](https://github.com/couchbase/couchbase-lite-core/blob/31266c807a208ffa1569a339fd1954dc0bbb7cb0/Java/jni/native_c4.cc#L148)